### ISSUE TITLE
M2 VCF to MAF conversion needs a command line

### ISSUE THREAD

**Issue by LeeTL1220** [broadinstitute/oncotator#313]
- ``TLOD`` (INFO) needs to be rendered as ``t_lod_fstar`` (not to be confused with the filter of the same name).
- ``--collapse-filter-cols`` must be specified!  This needs to go into Firehose.
- ``--skip-no-alt`` must be specified.
- ``AF`` (FORMAT) needs to be called tumor_f.
- ``AD`` annotation is comma separated (ref,alt), but in the MAF needs to be rendered as two separate values with (``t_ref_count`` and ``t_alt_count``).  Could also be array indexes, if need be (e.g. AD[0]).

Need to specify final command line and any code changes to support M2 VCFs being converted into MAFs that look like the ones in the current CGA FH workflow.

**Comment by LeeTL1220** [comment]
@kcibul Is the DREAM data public?  Can I use it for a unit test?

**Comment by LeeTL1220** [comment]
@kcibul some context: @ldgauthier says that the integration vcf is DREAM data.  I'd like to put it into the automated tests...

**Comment by kcibul** [comment]
yep -- that's legit!

-------------------------------
Kristian Cibulskis
Chief Architect, Data Sciences & Data Engineering
Broad Institute of MIT and Harvard
kcibul@broadinstitute.org
617-714-7637

**Comment by LeeTL1220** [comment]
@kcibul @ldgauthier So, I was wondering why (after a little bit of spent time chasing my tail) that integration_test_example.vcf does not have the ALT and REF orientation fields (e.g. ALT_F2R1)?  Can you guys provide a version of this vcf that does have those fields?

**Comment by LeeTL1220** [comment]
@kcibul @ldgauthier Do you guys want an alias for NLOD?  Such as ``n_lod``?  I see it in the VCF...

**Comment by LeeTL1220** [comment]
@kcibul @ldgauthier FYI... ``tumor_f`` will only come out when using ``-o TCGAMAF`` (the default and what will be the recommended standard for producing M2 MAFs)  I doubt this piece of information will ever matter, but I am mentioning it here.

**Comment by LeeTL1220** [comment]
@kcibul @ldgauthier Do you guys have a VCF from M2 with multiple alts on the same line?

**Comment by kcibul** [comment]
At the moment, I don't think we'll make calls like that.  Given that it's
single tumor-normal pair, it would mean calling a somatic mutation on top
of a germline SNP... which is a whole different calling model that we're
not supporting right now

-------------------------------
Kristian Cibulskis
Chief Architect, Data Sciences & Data Engineering
Broad Institute of MIT and Harvard
kcibul@broadinstitute.org
617-714-7637

**Comment by LeeTL1220** [comment]
I just worry how well it will play with existing functionality that handles
this.  Don't sweat it, I can construct a test myself.

On Tue, Jun 9, 2015 at 8:45 PM, kcibul <notifications@github.com> wrote:

> At the moment, I don't think we'll make calls like that. Given that it's
> single tumor-normal pair, it would mean calling a somatic mutation on top
> of a germline SNP... which is a whole different calling model that we're
> not supporting right now
>
> -------------------------------
> Kristian Cibulskis
> Chief Architect, Data Sciences & Data Engineering
> Broad Institute of MIT and Harvard
> kcibul@broadinstitute.org
> 617-714-7637
>
>
> On Tue, Jun 9, 2015 at 8:31 PM, Lee Lichtenstein <notifications@github.com
> >
> wrote:
>
> > @kcibul <https://github.com/kcibul> @ldgauthier
> > <https://github.com/ldgauthier> Do you guys have a VCF from M2 with
> > multiple alts on the same line?
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/broadinstitute/oncotator/issues/313#issuecomment-110544033
> >
>
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/broadinstitute/oncotator/issues/313#issuecomment-110547695>
> .
>



-- 
Lee Lichtenstein
Broad Institute
75 Ames Street, Room 7003EB
Cambridge, MA 02142
617 714 8632

**Comment by kcibul** [comment]
as for the integration tests with the ALT/REF F2R1 stuff -- @ldgauthier  can you run that from your branch?  I don't have that code

**Comment by kcibul** [comment]
n_lod as alias for NLOD sounds great, thanks

**Comment by LeeTL1220** [comment]
@kcibul I already have working unit tests that do the conversion above.  Let me check boxes for what has been implemented in my branch (and has a passing unit test).  All that is left is Firehose stuff, but I have not finished regression testing -- once I do that, I will assign this to you for FH.

**Comment by LeeTL1220** [comment]
@kcibul Please note that you may want to create a new task configuration for the new Oncotator.  It might make life easier to have separate parameters for tumor barcode and normal barcode in the task config itself.  Otherwise, MVEL will be necessary

**Comment by LeeTL1220** [comment]
@kcibul New task configuration and new module, since a hydrant.deploy change would be necessary to implement suggestion above..

**Comment by LeeTL1220** [comment]
@ldgauthier Is it possible for PGT to be different than GT?  (Other than ``/`` vs ``|``)

**Comment by LeeTL1220** [comment]
The checklist at the top of this issue represents changes in Firehose.  If something is not mentioned, then leave it the same as the current ``Oncotate {SNP, Indel} for {Capture, WGS}``

**Comment by LeeTL1220** [comment]
Solution for ``AD`` was fairly clunky.  The TcgaMafOutputRenderer has an option (always set to true) to split the ``allelic_depth`` annotation.  During VCF input, ``AD`` is mapped to the annotation, ``allelic_depth``.

**Comment by ldgauthier** [comment]
PGT could potentially be swapped vs. GT, i.e. GT = 0/1 and PGT = 1|0

**Comment by LeeTL1220** [comment]
Once pull request is complete, this issue should *not* be closed.

**Comment by LeeTL1220** [comment]
@kcibul 
v1.7.0.0.pre-alpha.1 was released

The code is on the VMs: ``/xchip/tcga/Tools/oncotator/releases/oncotator-1.7.0.0-pre-alpha.1``

See the Oncotator_v1 ``hydrant.deploy`` as a model for the module.  Note the ``build`` step in ``hydrant.deploy`` will only need a simple modification and you will need to adjust ``oncotator.sh`` as well (I believe).
